apps/api/src/campaign/campaign.service.ts

```diff
@@ -474,8 +474,8 @@ export class CampaignService {
     }
     //donation exists, so check if it is safe to update it
     else if (
-      donation?.status === newDonationStatus ||
-      donation?.status === getAllowedPreviousStatus(newDonationStatus)
+      donation?.status !== newDonationStatus &&
+      shouldAllowStatusChange(donation?.status, newDonationStatus)
```

Comment:
The previous implementation allowed the donation to be updated if the old and new status was the same. 
That was to allow replay of old events if we want to add more fields from the event.